client/app/components/WithIcon.tsx

```diff
@@ -0,0 +1,41 @@
+import React from "react";
+
+export interface WithIconProps {
+  children?: string;
+  icon: JSX.Element;
+  alt: string;
+}
+
+export function withIcon<ComponentProps extends object>(Component: React.ComponentType<ComponentProps>) {
+  return class WithIcon extends React.Component<ComponentProps & WithIconProps> {
+    render() {
+      const { children, icon, alt, ...props } = this.props;
+      return (
+        <Component {...(props as ComponentProps)}>
+          {children}
+          {children && " "}
+          {icon} <span className="sr-only">{alt}</span>
+        </Component>
+      );
+    }
+  };
+}
+
+/**
+ * Accessible wrapper for adding icons to a component.
+ * Defaults to the external icon.
```

Comment:
This is the most controversial part of these changes. I found it practical to have this available. @thielium please let me know your thoughts on this.